packages.go

```diff
@@ -0,0 +1,154 @@
+	SHA1      string     `json:"file_sha1"`
+}
+
+func (s PackageFile) String() string {
+	return Stringify(s)
+}
+
+// ListProjectPackagesOptions are the parameters available in a ListProjectPackages() Operation
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/packages.html#within-a-project
+type ListProjectPackagesOptions struct {
+	ListOptions
+	OrderBy            string `url:"order_by,omitempty" json:"order_by,omitempty"`
+	Sort               string `url:"sort,omitempty" json:"sort,omitempty"`
+	Type               string `url:"package_type,omitempty" json:"package_type,omitempty"`
+	Name               string `url:"package_name,omitempty" json:"package_name,omitempty"`
+	IncludeVersionless bool   `url:"include_versionless,omitempty" json:"include_versionless,omitempty"`
+}
+
+// ListProjectPackages gets a list of packages in a project.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/packages.html#within-a-project
+func (s *PackagesService) ListProjectPackages(pid interface{}, opt *ListProjectPackagesOptions, options ...RequestOptionFunc) ([]*Package, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/packages", pathEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var packages []*Package
+	resp, err := s.client.Do(req, &packages)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return packages, resp, err
+}
+
+// DeleteRegistryRepository deletes a repository in a registry.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/packages.html#delete-a-project-package
+func (s *PackagesService) DeleteProjectPackage(pid interface{}, packageID int, options ...RequestOptionFunc) (*Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprintf("projects/%s/packages/%d", pathEscape(project), packageID)
+
+	req, err := s.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
+
+// ListPackageFilesOptions represents the available
+// ListPackageFiles() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/packages.html#list-package-files
+type ListPackageFilesOptions ListOptions
+
+// ListPackageFiles gets a list of files that are within a package
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/packages.html#list-package-files
+func (s *PackagesService) ListPackageFiles(pid interface{}, packageID int, opt *ListPackageFilesOptions, options ...RequestOptionFunc) ([]*PackageFile, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/packages/%d/package_files",
+		pathEscape(project),
+		packageID,
+	)
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var tags []*PackageFile
```

Comment:
I guess `tags` should be `pfs`?

```suggestion
	var pfs []*PackageFile
```